sdk/ai/azure-ai-projects/README.md

```diff
@@ -683,6 +638,55 @@ with project_client.agents.create_stream(thread_id=thread.id, assistant_id=agent
 ```
 
 <!-- END SNIPPET -->
+
+In the code above, because an `event_handler` object is not passed to the `create_stream` function, the SDK will instantiate `AgentEventHandler` or `AsyncAgentEventHandler` as the default event handler and produce iterable with `event_type` and `event_data`.  `AgentEventHandler` and `AsyncAgentEventHandler` are overridable.  Here is an example:
+
+<!-- SNIPPET:sample_agents_stream_eventhandler.stream_event_handler -->
+
+```python
+class MyEventHandler(AgentEventHandler[str]):
+
+    def on_message_delta(self, delta: "MessageDeltaChunk") -> Optional[str]:
+        return f"Text delta received: {delta.text}"
+
+    def on_thread_message(self, message: "ThreadMessage") -> Optional[str]:
+        return f"ThreadMessage created. ID: {message.id}, Status: {message.status}"
+
+    def on_thread_run(self, run: "ThreadRun") -> Optional[str]:
+        return f"ThreadRun status: {run.status}"
+
+    def on_run_step(self, step: "RunStep") -> Optional[str]:
+        return f"RunStep type: {step.type}, Status: {step.status}"
+
+    def on_error(self, data: str) -> Optional[str]:
+        return f"An error occurred. Data: {data}"
+
+    def on_done(self) -> Optional[str]:
+        return "Stream completed."
+
+    def on_unhandled_event(self, event_type: str, event_data: Any) -> Optional[str]:
+        return f"Unhandled Event Type: {event_type}, Data: {event_data}"
+```
+
+<!-- END SNIPPET -->
+
+
+<!-- SNIPPET:sample_agents_stream_eventhandler.create_stream -->
+
+```python
+with project_client.agents.create_stream(
+    thread_id=thread.id, assistant_id=agent.id, event_handler=MyEventHandler()
+) as stream:
+    for event_type, event_data, func_return in stream:
```

Comment:
Do we document how customer can use the event function return? What is event function?

Reply:
After releasing, I will working on the file search sample app to use the event handler with return value.   Then I will add a link to this readme.